src/infra/dex/balancer/dto.rs

```diff
@@ -1,58 +1,208 @@
 use {
     crate::{
-        domain::{auction, dex, order},
+        domain::{dex, eth, order},
+        infra::dex::balancer::Error,
         util::serialize,
     },
+    bigdecimal::BigDecimal,
     ethereum_types::{H160, H256, U256},
+    number::conversions::{big_decimal_to_u256, u256_to_big_decimal},
     serde::{Deserialize, Serialize},
     serde_with::serde_as,
 };
 
-#[derive(Debug, Serialize)]
+pub mod get_swap_paths_query {
+    /// Get swap quote from the SOR v2 for the V2 vault.
+    pub const QUERY: &str = r#"
+query sorGetSwapPaths($callDataInput: GqlSwapCallDataInput!, $chain: GqlChain!, $queryBatchSwap: Boolean!, $swapAmount: AmountHumanReadable!, $swapType: GqlSorSwapType!, $tokenIn: String!, $tokenOut: String!, $useVaultVersion: Int) {
+    sorGetSwapPaths(
+        callDataInput: $callDataInput,
+        chain: $chain,
+        queryBatchSwap: $queryBatchSwap,
+        swapAmount: $swapAmount,
+        swapType: $swapType,
+        tokenIn: $tokenIn,
+        tokenOut: $tokenOut,
+        useVaultVersion: $useVaultVersion
+    ) {
+        tokenAddresses
+        swaps {
+            poolId
+            assetInIndex
+            assetOutIndex
+            amount
+            userData
+        }
+        swapAmountRaw
+        returnAmountRaw
+        tokenIn
+        tokenOut
+    }
+}
+    "#;
+}
+
+#[derive(Serialize)]
 #[serde(rename_all = "camelCase")]
-pub enum OrderKind {
-    Sell,
-    Buy,
+pub struct Query<'a> {
+    query: &'a str,
```

Comment:
Why to hold a reference to the query for every time we create this struct when the query cannot be selected and it will be always the same? why not to use it directly?